projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -461,18 +472,19 @@ export class IgxFilteringService implements OnDestroy {
         if (expressions instanceof FilteringExpressionsTree) {
             const expressionsTree = expressions as FilteringExpressionsTree;
             for (let i = 0; i < expressionsTree.filteringOperands.length; i++) {
-                this.generateExpressionsList(expressionsTree.filteringOperands[i], expressionsTree.operator, expressionsUIs);
+                this.generateExpressionsListRecursive(expressionsTree.filteringOperands[i], expressionsTree.operator, expressionsUIs);
+            }
+            if (expressionsUIs.length) {
```

Comment:
Isn't it better this check to be `expressionsUIs.length > 0`?

Reply:
It is not needed since 0 is a falsy value.